movement-test/src/main.c

```diff
@@ -69,6 +69,8 @@ void main(void) {
     UINT8 frame_skip = 8;  // Update player's animation every 8 frame to
                            // slow down the animation (8 frames = ~133 ms
                            // between each animation frames)
+    UINT8 dx = 0;
+    UINT8 dy = 0;
```

Comment:
It's debatable whether to make these global or not to save some stack space, but they should fit in registers...

Reply:
Other parts (like collisions!) will need these info ; they should be global (stored in a struct I think?)

Reply:
Ah yes, I hadn't considered that the dx/dy might be needed elsewhere. You're right.

I don't know about the structs. I'm a bit wary of alignment...